build/index.js

```diff
@@ -24,38 +41,90 @@ function downloadLib(name, path) {
   });
 }
 
-const libs = [];
-const cdnLibs = [];
-const paths = {};
-Object.entries(settings.paths).map(([name, path]) => {
-  if (/^(components|modules)\//.test(path)) {
-    return;
-  }
-  const isCdn = /^http(s?):\/\//.test(path);
-  if (isCdn) {
-    cdnLibs.push(downloadLib(name, path));
-  }
-  libs.push(name);
-  paths[name] = isCdn ? createCdnLibPath(name) : path;
-});
-
-settings = {
-  ...settings,
-  baseUrl: './js',
-  include: libs,
-  paths: {
-    ...paths,
-    'config-local': 'empty:',
-    'css-builder': 'empty:',
-  },
-  excludeShallow: [],
-  findNestedDependencies: false,
-  out: './js/assets/bundle/bundle.js',
-  optimize: 'uglify2',
-};
+function downloadLibs(aliasesMap, isJs = true) {
+  const libs = [];
+  const cdnLibs = [];
+  const paths = {};
+  Object.entries(aliasesMap).map(([name, path]) => {
+    if (/^(components|modules)\//.test(path)) {
+      return;
+    }
+    const isCdn = /^http(s?):\/\//.test(path);
+    if (isCdn) {
+      cdnLibs.push(downloadLib(name, path, isJs));
+    }
+    libs.push(name);
+    paths[name] = isCdn ? createCdnLibPath(name) : path;
+  });
 
-Promise.all(cdnLibs)
-  .then(() => {
-    console.log('Optimizing bundle...');
-    requirejs.optimize(settings, () => console.log('success'), (er) => console.error(er));
+  return {
+    libs,
+    cdnLibs,
+    paths,
+  };
+}
+
+function optimizeJs() {
+  const { libs, cdnLibs, paths } = downloadLibs(settings.paths);
```

Comment:
The function is called `optimizeJs` but not `downloadLibsAndOptimizeJs`. Let's keep things atomic and split the functionality.